classes/loginflow/authcode.php

```diff
@@ -639,6 +642,12 @@ protected function handlelogin(string $oidcuniqid, array $authparams, array $tok
                     throw new moodle_exception('errorauthloginfailednouser', 'auth_oidc', null, null, '1');
                 }
             }
+            if (!$CFG->allowaccountssameemail && array_key_exists('email', $userinfo)) {
+                // Here we know there is atleast 1 account with that email.
+                // We can just get the first match which is safe as long as allowaccountssameemail is off.
+                $user = $DB->get_record('user', ['email' => $userinfo['email']]);
+                $username = $user->username;
```

Comment:
I see that $user is used further below and think this can be more succinct and less variable pollute-y

You can use this API below to extract a single field, and set the $username field directly:
`$DB->get_field($table, $return, array $conditions, $strictness=IGNORE_MISSING)`
(Adjust the options to best fit the scenario)

Ref: https://docs.moodle.org/dev/Data_manipulation_API#get_field

Reply:
This was the result of a last min refactor. Updated.